src/components/molecule/Table.tsx

```diff
@@ -0,0 +1,54 @@
+import { tw } from '@/utils/tailwindMerge';
+import React from 'react';
+
+type TableProps<T extends React.ElementType> = Component<T> & {
+  column: Array<{ name: string; key: string }>;
+  data: Array<object>;
+};
+
+type TheadProps = {
+  column: Array<{ name: string; key: string }>;
+};
+
+export function Table({ column, data, className }: TableProps<'table'>) {
+  return (
+    <table className={tw('table-auto w-full text-m text-center', className)}>
+      <Thead column={column}></Thead>
+      <Tbody column={column} data={data}></Tbody>
+    </table>
+  );
+}
+
+function Thead({ column }: TheadProps) {
+  return (
+    <thead className="text-m uppercase bg-primary-200">
+      <tr>
+        {column.map(({ name, key }) => {
+          return (
+            <th scope="col" className="py-3" key={key}>
+              {name}
+            </th>
+          );
+        })}
+      </tr>
+    </thead>
+  );
+}
+
+function Tbody({ column, data }: TableProps<'table'>) {
+  return (
+    <tbody className="bg-white border-b">
+      {Object.entries(data).map(([key, value]) => (
+        <tr key={key} className="border-b border-primary-100">
+          {column.map(({ key }) => {
+            return (
+              <td className="py-3" key={key}>
+                {value[key as keyof typeof value]}
+              </td>
+            );
+          })}
+        </tr>
+      ))}
+    </tbody>
```

Comment:
또 onclick 관련해서, tr(열) 또는 td(셀)를 누르면 click 이벤트가 작동하도록 추가하면 될 거 같습니다.
아래의 코드 예시를 참고하여 제안합니다🤔
```suggestion
      <tbody {...getTableBodyProps()}>
        {rows.map((row, i) => {
          prepareRow(row);
          return (
            <tr {...row.getRowProps()} onClick={() => console.log(row.original)}>
              {row.cells.map((cell, j) => {
                return (
                  <td
                    rowSpan={cell.rowSpan}
                    {...cell.getCellProps()}
                    style={borderStyle}
                  >
                    {cell.render("Cell")}
                  </td>
                );
              })}
            </tr>
          );
        })}
      </tbody>
```
* 참고 자료: https://github.com/TanStack/table/discussions/2295

Reply:
의견 감사합니다.

제 고민은 row마다 이동해야하는 페이지가 다른데 data라는 array props 안에 객체마다 onClick을 가지고 있어야할지 아니면 그냥 하나의 onClick으로 row의 id값만 넘기면 될지에 대한 고민이었습니다.

그런데 지금 생각해보니 row id값을 넘기면 되겠다는 생각이 드네요.

Reply:
넵넵 저도 똑같이 tr(row 전체)에 onclick 이벤트를 걸어서 해당 row 페이지로 가는 방향으로 생각했어요. 예를 들어 위 코드에서는
* <tr {...row.getRowProps()} onClick={해당 열로 라우팅해주는 함수}>
* 라우팅해주는 함수에는 row의 id 값을 넘겨주도록 할 것
위와 같은 방식으로 생각했습니다.